pkg/didcomm/protocol/issuecredential/states.go

```diff
@@ -234,11 +234,17 @@ func (s *requestReceived) CanTransitionTo(st state) bool {
 }
 
 func (s *requestReceived) ExecuteInbound(md *metaData) (state, stateAction, error) {
+	var req = RequestCredential{}
+	if err := md.Msg.Decode(&req); err != nil {
+		return nil, nil, fmt.Errorf("decode: %w", err)
+	}
+
 	// creates the state's action
 	action := func(messenger service.Messenger) error {
 		// sets message type
 		return messenger.ReplyTo(md.Msg.ID(), service.NewDIDCommMsgMap(IssueCredential{
-			Type: IssueCredentialMsgType,
+			Type:              IssueCredentialMsgType,
+			CredentialsAttach: req.RequestsAttach,
```

Comment:
The attachment should not be replayed back to the sender of the request. This new attachment should contain a credential.

Reply:
Done! I have added `WithIssueCredential` function.